var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -69,18 +69,25 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("boost@1.79.0:", when="@2.2:")
     depends_on("kokkos@3.2.00:", when="+kokkos")
     depends_on("kokkos@3.7:", when="+kokkos @2.3:")
+    depends_on("kokkos@3.7:", when="@2.4:")
     depends_on("kokkos +cuda +cuda_constexpr +cuda_lambda", when="+kokkos +cuda")
+    requires("^kokkos +cuda_constexpr +cuda_lambda", when="^kokkos +cuda")
     depends_on("kokkos +rocm", when="+kokkos +rocm")
     depends_on("kokkos +openmp", when="+kokkos +openmp")
     depends_on("legion@cr-20210122", when="backend=legion @2.0:2.2.1")
     depends_on("legion@cr-20230307", when="backend=legion @2.2.0:2.2.1")
     depends_on("legion@24.03.0:", when="backend=legion @2.2.2:")
+    depends_on("legion@24.09.0:", when="backend=legion @2.3.1:")
     depends_on("legion+shared", when="backend=legion +shared")
     depends_on("legion+hdf5", when="backend=legion +hdf5")
     depends_on("legion+kokkos", when="backend=legion +kokkos")
     depends_on("legion+openmp", when="backend=legion +openmp")
     depends_on("legion+cuda", when="backend=legion +cuda")
     depends_on("legion+rocm", when="backend=legion +rocm")
```

Comment:
Aren't these superseded by the new `depends_on` below that reacts to Kokkos' existence (perhaps required by `+kokkos`) or configuration?

Reply:
Fixed.